### PR TITLE
Add RubyGems Monthly Update for July 2023

### PR DESCRIPTION
**Pull request by lauragift21**
RubyGems monthly update roundup

### CODE REVIEW

**Comment by lauragift21** [comment]
Hi @deivid-rodriguez, Could you help review this month's update, let me know if there's anything I'm missing.

**Review by deivid-rodriguez — COMMENTED** [review]
From the list of improvements, some of them were not actually included in the releases mentioned in the first paragraph while others were just follow up to other PRs.

I took the liberty to reword the stuff after the first paragraph to make it a bit more accurate and representative of the work:

> In Bundler, the above releases include several improvements such as [fixes](https://github.com/rubygems/rubygems/pull/6784) related to locked platforms, [improvements](https://github.com/rubygems/rubygems/pull/6786) to make sections related to git gems stable, and some other regression fixes.
>
> In RubyGems, we [improved](https://github.com/rubygems/rubygems/pull/6810) certain `gem install` invocations that had gotten slower since the dependencies API was phased out.
>
> Some other improvements that landed into our repo this month but that are not included in the above releases are:
>
> * Improvements to RubyGems require - [#6827](https://github.com/rubygems/rubygems/pull/6827).
> * Improvements related to the [Security Devices support feature](https://blog.rubygems.org/2023/08/03/level-up-using-security-devices.html), such as [#6774](https://github.com/rubygems/rubygems/pull/6774).

**Comment by lauragift21** [comment]
> From the list of improvements, some of them were not actually included in the releases mentioned in the first paragraph while others were just follow up to other PRs.
> 
> I took the liberty to reword the stuff after the first paragraph to make it a bit more accurate and representative of the work:
> 
> > In Bundler, the above releases include several improvements such as [fixes](https://github.com/rubygems/rubygems/pull/6784) related to locked platforms, [improvements](https://github.com/rubygems/rubygems/pull/6786) to make sections related to git gems stable, and some other regression fixes.
> > In RubyGems, we [improved](https://github.com/rubygems/rubygems/pull/6810) certain `gem install` invocations that had gotten slower since the dependencies API was phased out.
> > Some other improvements that landed into our repo this month but that are not included in the above releases are:
> > 
> > * Improvements to RubyGems require - [#6827](https://github.com/rubygems/rubygems/pull/6827).
> > * Improvements related to the [Security Devices support feature](https://blog.rubygems.org/2023/08/03/level-up-using-security-devices.html), such as [#6774](https://github.com/rubygems/rubygems/pull/6774).

Thank you, @deivid-rodriguez. Do the other items on the list still apply to the improvements that were added to the repository, or can I remove them as well?

**Comment by deivid-rodriguez** [comment]
Good point, we can probably keep them I think, except for 

> - set thread configurations in the thread-local manner - [#6842](https://github.com/rubygems/rubygems/pull/6842).
> - restored support for Pathname objects in the replaced require - [#6837](https://github.com/rubygems/rubygems/pull/6837).

which are replaced but the two items I listed (they are follow up to those improvements).

**Comment by lauragift21** [comment]
Got it thanks!

> Good point, we can probably keep them I think, except for
> 
> > * set thread configurations in the thread-local manner - [#6842](https://github.com/rubygems/rubygems/pull/6842).
> > * restored support for Pathname objects in the replaced require - [#6837](https://github.com/rubygems/rubygems/pull/6837).
> 
> which are replaced but the two items I listed (they are follow up to those improvements).

Got it thanks!

File: _posts/2023-08-07-july-rubygems-updates.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+- improved RubyGems `require` - [#6827](https://github.com/rubygems/rubygems/pull/6827).
+- improvements related to the Security Devices support feature - [#6774](https://github.com/rubygems/rubygems/pull/6774).
+- updated Magnus version in Rust extension gem template - [#6843](https://github.com/rubygems/rubygems/pull/6843).
+- removed side effects when unmarshaling old `Gem::Specification` - [#6825](https://github.com/rubygems/rubygems/pull/6825).
+- used `File::NULL `instead of hard coded null device names - [#6809](https://github.com/rubygems/rubygems/pull/6809).
+- added Automatiek improvements - [#6788](https://github.com/rubygems/rubygems/pull/6788).
+- fixed another incorrect removal of "Ruby" platform from lockfile when changing path sources - [#6784](https://github.com/rubygems/rubygems/pull/6784).
+- fixed git source lockfile instability - [#6786](https://github.com/rubygems/rubygems/pull/6786).
+- excluded Bundler from missing locked dependencies check - [#6792](https://github.com/rubygems/rubygems/pull/6792).
```
Comment (deivid-rodriguez) [inline _posts/2023-08-07-july-rubygems-updates.md:26]: Sorry for the back and forth @lauragift21. Actually last three entries were included in the above releases, so not necessary to list here.

And I actually meant to link to a couple of them in the second paragraph:

`In Bundler, the above releases include several improvements such as [fixes](https://github.com/rubygems/rubygems/pull/6784) related to locked platforms, [improvements](https://github.com/rubygems/rubygems/pull/6786) to make sections related to git gems stable, and some other regression fixes.`

Comment (lauragift21) [reply _posts/2023-08-07-july-rubygems-updates.md:26]: No problem at all, I'll update the post.

**Review by deivid-rodriguez — APPROVED** [review]
Awesome!

**Pull request merged**